lib/layer.js

```diff
@@ -41,10 +40,27 @@ function Layer (path, options, fn) {
   this.name = fn.name || '<anonymous>'
   this.params = undefined
   this.path = undefined
-  this.regexp = pathRegexp((opts.strict ? path : loosen(path)), this.keys, opts)
-
-  // set fast path flags
-  this.regexp._slash = path === '/' && opts.end === false
+  this.slash = path === '/' && opts.end === false
+
+  var matchers = []
+
+  function addMatcher (_path) {
+    var matcher = pathRegexp.match((opts.strict ? _path : loosen(_path)), {
+      sensitive: opts.sensitive,
+      end: opts.end,
+      trailing: !opts.strict,
+      decode: decodeParam
+    })
+    matchers.push(matcher)
```

Comment:
Stylistic, but not a huge fan of code with side effects. I think this would be better written as: 

```js
function matcher(path) { return pathToRegexp.match(...); }

const matchers = Array.isArray(path) ? path.map(matcher) : [matcher(path)];
```

Reply:
I also did not like this code lol, but it was my first attempt and it worked so I was tired so took a nap instead of changing it. But I agree and will update it.